src/interface/report/Results/Timeline/Component.js

```diff
@@ -109,6 +109,18 @@ class Timeline extends React.PureComponent {
     return eventsBySpellId;
   }
 
+  _getCanonicalId(spellId){
```

Comment:
This seems sensible, though not 100% sure it's without problems.

Reply:
it's the method used in SpellUsable.js to track shared cooldowns so I figured it'd be safe to use here too since the timeline should group abilities by shared cooldowns too I think